src/include/kdbopmphmpredictor.h

```diff
@@ -37,6 +37,7 @@ typedef struct
 	uint8_t * patternTable; /*!< the global pattern history table */
 	size_t size;		/*!< size of patternTable in bytes */
 	size_t lookupCount;     /*!< number of lookups made without alteration of the KeySet */
+	size_t ksSize;		/*!< number of keys in the KeySet */
```

Comment:
What is different here to the size of the keyset?

Reply:
The predictor generally looks at old events to predict actual once.
At the prediction the keyset size is need to determine if it **was** worth using the OPMPHM or not. But at the prediction the keyset size may has changed, therefore the old one is needed.

Reply:
Thank you. This information should be in the comment.